src/phase0/state_transition/mod.rs

```diff
@@ -106,16 +141,50 @@ pub fn is_valid_indexed_attestation<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    indexed_attestation: IndexedAttestation<MAX_VALIDATORS_PER_COMMITTEE>,
+    indexed_attestation: &IndexedAttestation<MAX_VALIDATORS_PER_COMMITTEE>,
     context: &Context,
 ) -> Result<(), Error> {
-    if indexed_attestation.attesting_indices.is_empty() {
-        return Err(Error::InvalidOperation);
+    let attesting_indices = &indexed_attestation.attesting_indices;
+    if attesting_indices.is_empty() {
+        return Err(Error::InvalidOperation(
+            InvalidOperation::IndexedAttestation(InvalidIndexedAttestation::AttestingIndicesEmpty),
+        ));
     }
-    let indices: HashSet<usize> =
-        HashSet::from_iter(indexed_attestation.attesting_indices.iter().cloned());
+    let check_sorted = |list: &[ValidatorIndex]| -> Result<(), Error> {
+        list.iter()
+            .tuple_windows()
+            .enumerate()
+            .try_for_each(|(_, (x, y))| {
+                if x < y {
+                    Ok(())
+                } else {
+                    Err(Error::InvalidOperation(
+                        InvalidOperation::IndexedAttestation(
+                            InvalidIndexedAttestation::AttestingIndicesNotSorted,
+                        ),
+                    ))
+                }
+            })?;
+        Ok(())
+    };
+    check_sorted(attesting_indices.as_slice())?;
```

Comment:
i think this can be done more simply and we can drop the dependency on `itertools`:

something like:

```rust
let is_sorted = attesting_indices.windows(2).map(|pair| { let a = &pair[0]; let b = &pair[1]; a < b }).all(|x| x);
if !is_sorted {
    return Err(...)
}
```

Reply:
ok